### ISSUE TITLE
QGIS build with Qt6  as 'experimental/preview build' for Linux users?

### ISSUE THREAD

**Issue by rduivenvoorde** [flathub/org.qgis.qgis#428]
Probably a lot of work (?), and I'm not sure if it is even possible, because I do not know if there is a 'base' for it.

But, 

QGIS is discussing to maybe make a Qt6 build of 3.36 available:
https://lists.osgeo.org/pipermail/qgis-developer/2024-February/066466.html

But building QGIS for Qt6 is hard, as (for example) Debian (even testing) is still missing some (packaged) dependencies.

I was wondering, IF even possible, if it would be doable to create a Flatpak version of this Qt6 build of 3.36.

In this way all flatpak users could have a look?

Though I also realize, that there will probably be a lot of gui/ui/gnome issues anyway, and maybe users will blame flatpak for that :-(

Any thoughts?

**Comment by aleixpol** [comment]
If all the dependencies are released, I don't see why we wouldn't. What would be the use-case? for developers? beta testing?

**Comment by rduivenvoorde** [comment]
Yes, exactly that. And for Linux users eager to 'see QGIS based on Qt6'.

Also for python/plugin developers who need to update the python code for the Qt6 move.

Maybe wait a little time until the QGIS actually is building/releasing previews itself.

**Comment by rduivenvoorde** [comment]
Mmm, thought to check if I would be able to change the manifest, by first use current version and build that one.

But failing because: 
- a untar/zip issue with cfitsio (and another sha)
- another mariadb download (fixed, and downloaded)
- then a mariadb build failure on: CMake Error at cmake/libfmt.cmake:44 (MESSAGE):  system libfmt library is not found or unusable

Should it be possible to 'just compile stuff' on your host machine? Or do you better have a dedicated builder and maybe flatpak-builder version?

**Comment by aleixpol** [comment]
I build locally all the time, it should work fine.

**Comment by rduivenvoorde** [comment]
Would it be an idea to add a 'how to build the flatpak' instructions to readme.md?

Then I could try those :-)

**Comment by rduivenvoorde** [comment]
I was finally able to do a flatpak Qt6 build \o/

see: [org.qgis.qgis.json](https://github.com/user-attachments/files/20630734/org.qgis.qgis.json)

![Image](https://github.com/user-attachments/assets/546c42b0-0a2c-4e54-a93e-d32480d7c827)

If others want to try, first do a local build on your laptop of current stable version from THIS repo:

```
# make sure you have flatpak and flatpak-builder
# clone flatpub repo
git clone git@github.com:flathub/org.qgis.qgis.git
# get shared modules 
git clone --recurse-submodules git@github.com:flathub/org.qgis.qgis.git
# then do the actual build:
cd org.qgis.qgis
flatpak-builder --force-clean --user --install-deps-from=flathub --repo=repo --install builddir org.qgis.qgis.json
# after build
flatpak run org.qgis.qgis
```

I did have some hickups (downloading of sources sometimes fail and upon a second time just succeed).

In current Qt5 version you will end up with sip errors, for that reason I (hacky) changed the qgis module part in the org.qgis.qgis.json to fetch current master from git (where this is fixed)...

After my succesfull Qt5 build, I changed the base-version and runtime-version to Qt6 versions and then module by module fixed small things needed to be able to build Qt6 versions of those.

@aleixpol is there a proper way to distribute such builds? I was hoping for some 'export' function (like an AppImage, or a container tar or so...). So people could download a tar/zip, unpack and run?

OR should I just serve current build 'repo' dir on a webserver?

Or is the proper way to do a Qt6 test project? as org.qgis.qgis.qt6test or so ??

Any hints appreciated.

And thanks for all the work you already put in current stable build! There is an awful list of things that can go wrong :-)

**Comment by kannes** [comment]
You rock, @rduivenvoorde!

I just thought I'd give it a try to compile myself a fresh version but it looks like the `flatpak-builder` step pulls an ungodly amount of complex external library repos and I assume it will have to compile them all? That would be total overkill for me sadly.

I guess that's just how it works but thought I'd ask before I do something stupid 🤡

**Comment by aleixpol** [comment]
Could you create a pull request? we can rebase it until upstream decides Qt 6 is the way to go. It's about time too.
